designdocs/MESSAGE_STORAGE.md

```diff
@@ -110,13 +99,22 @@ from `StartingBlockNumber` until is reaches one of values from `EndBlockNumber`
     -  nextBlock: `Optional<BlockNumber>` (has value if hasNext is true)
     -  nextPage: `Optional<u32>` (has value if hasNext is true)
 
-#### Cleanup (Retention policy)
+#### Cleanup (Retention policy - heavy)
 1. `on_initialize` remove all values from **Messages** for all blocks in following range
 [`StartingBlock`, currentBlock - `StoragePeriod` - 1]
 2. update `StartingBlock` to `max(StartingBlock, currentBlock - StoragePeriod - 1 ) + 1`
 3. `on_initialize` calculate and return Weight for number of database read and writes for
 `on_initialize` +  `on_finalize`
 
+#### Cleanup (Retention policy - light)
+The idea is to implement inserts using the same concept as a circular array. We will calculate a
+`target_block` which is going to be `current_block % retention_policy_block_size` and insert
+messages into this calculated target block. This will allow us to override messages which are
+already outside retention policy with new ones.
+
+Note: There are going to be some left-over blocks and messages that still needs another process to
+clean up.
+
```

Comment:
Some clarity around what are left over blocks and their significance, say along with this note